go/consensus/cometbft/apps/roothash/state/state.go

```diff
@@ -51,6 +51,11 @@ var (
 	//
 	// Value is CBOR-serialized message.IncomingMessage.
 	inMsgQueueKeyFmt = keyformat.New(0x29, keyformat.H(&common.Namespace{}), uint64(0))
+	// Previous state and I/O runtime roots.
+	// Key format is: 0x2a H(<runtime-id>) <round>
+	// Value is CBOR-serialized array of past N state and I/O roots.
```

Comment:
This is not true, the value is just the array containing state and I/O roots for the given round, in order. I would rather define something like the following (in the roothash API) and serialize that:
```go
type PastRoots struct {
	_ struct{} `cbor:",toarray"` //nolint

	StateRoot hash.Hash
	IORoot    hash.Hash
}
```
This gets serialized as `[]hash.Hash{StateRoot, IORoot}`.

Reply:
Yeah, I know, it's an old comment, I was just in the process of updating it :)